### PR TITLE
Remove GenericWork from the views

### PR DESCRIPTION
**Pull request by jcoyne**
This positions us to offer more work types besides `GenericWork`
@projecthydra/sufia-code-reviewers

### CODE REVIEW

**Comment by mjgiarlo** [comment]
Refs #2025.

File: app/views/sufia/homepage/_home_header.html.erb [hunk 1]
```diff
@@ -1,13 +1,13 @@
 <div class="home_call_action col-xs-12 col-sm-4 pull-right">
-  <% if can? :view_share_work, GenericWork %>
+  <% if Sufia.config.always_display_share_button || can_ever_create_works? %>
```
Comment (jeremyf) [inline app/views/sufia/homepage/_home_header.html.erb]: Not a fan of this logic; Where is `can_ever_create_works?` defined? It makes sense to me to have a method that reflects this particular bit of logic?

Comment (jcoyne) [inline app/views/sufia/homepage/_home_header.html.erb]: The idea here is that sufia by default has a "Share your work" button on the homepage. Some institiutions (Oregon Digital) don't want this turned on unless the user actually has permission. `can_ever_create_works?` is from CurationConcerns (originally from Curate I think).

Comment (jeremyf) [inline app/views/sufia/homepage/_home_header.html.erb]: I was wondering about consolidating the following into a single method:

```ruby```
Sufia.config.always_display_share_button || can_ever_create_works?
```

There are two layers of abstraction; A method call and a configuration option co-mingled in the view.

Comment (jcoyne) [inline app/views/sufia/homepage/_home_header.html.erb]: 👌

**Pull request merged**